### PR TITLE
improve compilation

### PR DESCRIPTION
**Pull request by abcastro**
tested on macos and linux

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,22 +10,30 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
-set(COVERAGE "Coverage" ON)
+option(COVERAGE "Coverage" ON)
```
Comment (abcastro) [inline CMakeLists.txt:13]: so you can do `cmake -DCOVERAGE=OFF`

File: CMakeLists.txt [hunk 2]
```diff
@@ -10,22 +10,30 @@ set(CMAKE_CXX_STANDARD 20)
...
+
 set(COMPILE_FLAGS -Wall -Werror -fsanitize=address -fsanitize=leak -pthread)
 set(LINK_FLAGS -fsanitize=address)
 
 #SET(GCC_WERR_COMPILE_FLAGS "-Wall -Werror")
 #SET(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} ${GCC_WERR_COMPILE_FLAGS} -fsanitize=address")
 #SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
 
+include_directories(${libwebqc_SOURCE_DIR} ${libwebqc_SOURCE_DIR}/include)
```
Comment (abcastro) [inline CMakeLists.txt:22]: for the `#include "../include/XXX.h`

File: CMakeLists.txt [hunk 3]
```diff
@@ -10,22 +10,30 @@ set(CMAKE_CXX_STANDARD 20)
...
+
+find_package(PkgConfig)
+
 find_package(CURL REQUIRED)
 include_directories(${CURL_INCLUDE_DIR})
 
+pkg_check_modules(CJSON REQUIRED libcjson)
```
Comment (abcastro) [inline CMakeLists.txt:29]: auto locate dependencies

File: include/webqc-curl.h [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #pragma once
 #include <stdbool.h>
-#include "../libwebqc.h"
```
Comment (abcastro) [inline include/webqc-curl.h:3]: I'm assuming it's nicer to not do like this

**Comment by urysegal** [comment]
Hi Arthur can you re-do this PR - I did not see it before I merged mine :-(

**Comment by urysegal** [comment]
I have put all this into master except one thing (see comment above) so I am closing this.

File: CMakeLists.txt (hunk 3 above)
Comment (urysegal) [reply CMakeLists.txt:29]: DOesn't work on  my 18.04 Ubuntu, they only added in 20.something

**Pull request closed without merging**